src/main/java/mango/Duke.java

```diff
@@ -0,0 +1,76 @@
+package mango;
+
+import java.io.IOException;
+
+/**
+ * Represents a chat-bot named Mango that keeps track of a list of tasks that the user
+ * can manipulate by adding, deleting, or completing.
+ */
+public class Duke {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructor for a Duke chat-bot.
+     */
+    public Duke() {
+        ui = new Ui();
+        storage = new Storage("data/mango.txt");
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (IOException e) {
+            ui.showLoadingError(new DukeException(e.getMessage()));
+            tasks = new TaskList();
+        }
+    }
+    /**
+     * Constructor for a Duke chat-bot.
+     *
+     * @param filePath The path for the file that will contain the list of tasks tracked by the chat-bot.
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (IOException e) {
+            ui.showLoadingError(new DukeException(e.getMessage()));
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Returns a greeting from the ui.
+     *
+     * @return A greeting message to the user.
+     */
+    public String greet() {
+        return this.ui.greet();
+    }
+
+    /**
+     * Returns a farewell message from the ui.
+     *
+     * @return A farewell message to the user.
+     */
+    public String exit() {
+        try {
+            this.storage.save(this.tasks);
```

Comment:
Honestly, I just really like this method of saving to file at the end instead of constantly, nice!

Reply:
This might be a problem if the application crashes (due to an exception) while the user is using it. All the tasks created then will not be saved. There is a forum post on this.